### ISSUE TITLE
How about transforming qmake sources into cmake/meson/bake ones?

### ISSUE THREAD

**Issue by KOLANICH** [eraxillan/qmake-ng#1]

**Comment by eraxillan** [comment]
Hi @KOLANICH, thanks for your interest :)
Well, there are some attempts to this task: e.g. https://code.qt.io/cgit/qt/qtbase.git/tree/util/cmake/pro2cmake.py?h=wip/cmake

I see qmake-ng as qbs-like tool: interpret statements directly, without generation of any makefiles/ninja-files/etc.
However, do you think it is pointless?

**Comment by KOLANICH** [comment]
>However, do you think it is pointless?

I don't think it is pointless. I don't have any strong opinion about it because I haven't used qmake in my projects. But qmake seems to be too low level at first sight in the sense I have to manually specify flags for C++ version instead of just specifying the version and CMake does the rest. I am not professional in Qmake, but it seems that CMake has a lot more batteries.

**Comment by eraxillan** [comment]
qmake of course much more low-level system than cmake.
However, it still have some stuff: [CONFIG variable](https://doc.qt.io/qt-5/qmake-variable-reference.html#config)
E.g. you can switch C++ standard, all warnings, debug/release build, add icon for macOS and some more.

So, if you don't use qmake, why you interested in it's projects conversion? :)
Is it just a "what if..." thought?

Because i'm afraid writing qmake <--> cmake converter will be extremely time-consuming task due to need to learn ~~ugly~~ cmake syntax too.

**Comment by KOLANICH** [comment]
>However, it still have some stuff: CONFIG variable

Thanks, that's cool. I probably should start using it.

>So, if you don't use qmake, why you interested in it's projects conversion? :)

To read variables from `pri` files into CMake.

**Comment by eraxillan** [comment]
qmake is still pretty handy for small projects: that's why i'm still interesting to make it better.
However, more significant is "just for fun" reason :)

About pri-files reading.
qmake and CMake both generate Makefile (for Unix-like systems).
You can avoid pro/pri-files parsing by using [CMake ExternalProject](https://cmake.org/cmake/help/latest/module/ExternalProject.html).
Is it an option?
Or you need exactly extracting some variables values from pro/pri instead of full compile?

**Comment by KOLANICH** [comment]
Yes, I want to extract variables from `.pri` and use their values in CMake, doing everything other with CMake.

>You can avoid pro/pri-files parsing by using CMake ExternalProject.
Is it an option?

I probably can.

**Comment by eraxillan** [comment]
Well, i forgot to write a final answer.
To properly implement variable extraction from pro/pri-files,
i need implement full eval first.
Currently it is in alpha stage and cannot process qmake mkspecs for example.
When (and if) pro/pri evaluation will be completed, i'll just add appropriate command line
option for your task.

**Comment by KOLANICH** [comment]
Thanks.
